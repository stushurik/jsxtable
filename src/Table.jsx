// @flow

import React, { type ElementType, type Node } from 'react';

export type Row = {
  [prop: string]: string,
  selected?: boolean,
  renderCell?: (row: Row, header: string) => ?Node
};

export type Data = Row[];

export type Header = {
  name: string,
  render?: (header: string) => ?Node
};

export type Props = {
  data: Data,
  className?: string,
  tableTag?: ?ElementType,
  trTag?: ?ElementType,
  tdTag?: ?ElementType,
  thTag?: ?ElementType,
  theadTag?: ?ElementType,
  tbodyTag?: ?ElementType,
  headers?: Header[],
  renderSelector?: (selected: boolean) => ?Node,
  onHeaderClick?: (header: string, position: number) => void,
  onDataClick?: (
    column: number,
    row: number,
    header: string,
    value: string
  ) => void,
  onRowSelect?: (row: Row, position: number) => void
};

export default function(props: Props): Node {
  const Table = props.tableTag || 'table';
  const Thead = props.theadTag || 'thead';
  const Tbody = props.tbodyTag || 'tbody';
  const Tr = props.trTag || 'tr';
  const Td = props.tdTag || 'td';
  const Th = props.thTag || 'th';

  const headers = props.headers || sniffHeaders(props.data);

  const onHeaderClick = props.onHeaderClick || noop;
  const onDataClick = props.onDataClick || noop;
  const onRowSelect = props.onRowSelect || noop;

  return (
    <Table className={props.className}>
      <Thead>
        <Tr className="headerRow">
          {headers.map((header, i) => (
            <Th
              className="header"
              key={`${i}${header.name}`}
              tid="header"
              onClick={() => onHeaderClick(header.name, i)}
            >
              {header.render ? header.render(header.name) : header.name}
            </Th>
          ))}
        </Tr>
      </Thead>
      <Tbody>
        {props.data.map((row: Row, i) => (
          <Tr
            className="dataRow"
            key={i}
            {...(props.renderSelector && props.onRowSelect
              ? {}
              : {
                  tid: 'rowSelector',
                  onClick: () => onRowSelect(row, i)
                })}
          >
            {props.renderSelector && (
              <Td tid="rowSelector" onClick={() => onRowSelect(row, i)}>
                {props.renderSelector(!!row.selected)}
              </Td>
            )}
            {headers.map((header, j) => (
              <Td
                className="data"
                key={`${i}${header.name}`}
                tid="data"
                onClick={() => onDataClick(j, i, header.name, row[header.name])}
              >
                {row.renderCell
                  ? row.renderCell(row, header.name)
                  : row[header.name]}
              </Td>
            ))}
          </Tr>
        ))}
      </Tbody>
    </Table>
  );
}

function sniffHeaders(data: Data): Header[] {
  return [
    ...data
      .reduce((set: Set<string>, row: Row) => {
        Object.keys(row)
          .filter(
            (column: string) => column !== 'renderCell' && column !== 'selected'
          )
          .forEach((column: string) => set.add(column));
        return set;
      }, new Set())
      .values()
  ].map(name => ({
    name
  }));
}

function noop(..._args: any[]) {}
