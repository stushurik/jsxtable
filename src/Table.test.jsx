import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Table from './Table';

describe('table', () => {
  it('should render table using default tags', () => {
    const data = [
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '23', Name: 'Lee Salminen' },
      { Age: '28', Position: 'Developer' },
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '30', Name: 'Test Person' },
      { Name: 'Another Test', Age: '26', Position: 'Developer' },
      { Name: 'Third Test', Age: '19', Position: 'Salesperson' },
      { Age: '23', Name: 'End of this Page', Position: 'CEO' }
    ];

    const props = { data };

    const wrapper = renderer.create(<Table {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render table using custom tags', () => {
    const data = [
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '23', Name: 'Lee Salminen' },
      { Age: '28', Position: 'Developer' },
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '30', Name: 'Test Person' },
      { Name: 'Another Test', Age: '26', Position: 'Developer' },
      { Name: 'Third Test', Age: '19', Position: 'Salesperson' },
      { Age: '23', Name: 'End of this Page', Position: 'CEO' }
    ];

    const tableTag = 'myAwesomeTable';
    const trTag = 'myAwesomeTr';
    const tdTag = 'myAwesomeTd';
    const thTag = 'myAwesomeTh';
    const theadTag = 'myAwesomeThead';
    const tbodyTag = 'myAwesomeTbody';

    const props = { data, tableTag, trTag, tdTag, thTag, theadTag, tbodyTag };

    const wrapper = renderer.create(<Table {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should alow to render row with customization', () => {
    const data = [
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '23', Name: 'Lee Salminen' },
      { Age: '28', Position: 'Developer' },
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '30', Name: 'Test Person' },
      { Name: 'Another Test', Age: '26', Position: 'Developer' },
      { Name: 'Third Test', Age: '19', Position: 'Salesperson' },
      {
        Age: '23',
        Name: 'End of this Page',
        Position: 'CEO',
        renderCell: (row, columnName) => <span>{row[columnName]}</span>
      }
    ];

    const props = { data };

    const wrapper = renderer.create(<Table {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should to set headers order', () => {
    const data = [
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '23', Name: 'Lee Salminen' },
      { Age: '28', Position: 'Developer' },
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '30', Name: 'Test Person' },
      { Name: 'Another Test', Age: '26', Position: 'Developer' },
      { Name: 'Third Test', Age: '19', Position: 'Salesperson' },
      {
        Age: '23',
        Name: 'End of this Page',
        Position: 'CEO',
        render: (row, columnName) => <span>{row[columnName]}</span>
      }
    ];

    const props = {
      data,
      headers: [{ name: 'Position' }, { name: 'Name' }, { name: 'Age' }]
    };

    const wrapper = renderer.create(<Table {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should alow to render header with customization', () => {
    const data = [
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '23', Name: 'Lee Salminen' },
      { Age: '28', Position: 'Developer' },
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '30', Name: 'Test Person' },
      { Name: 'Another Test', Age: '26', Position: 'Developer' },
      { Name: 'Third Test', Age: '19', Position: 'Salesperson' },
      {
        Age: '23',
        Name: 'End of this Page',
        Position: 'CEO',
        render: (row, columnName) => <span>{row[columnName]}</span>
      }
    ];

    const props = {
      data,
      headers: [
        {
          name: 'Position',
          render: header => <span>{header}</span>
        },
        { name: 'Name' },
        { name: 'Age' }
      ]
    };

    const wrapper = renderer.create(<Table {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should allow to render row selector', () => {
    const data = [
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '23', Name: 'Lee Salminen' },
      { Age: '28', Position: 'Developer' },
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '30', Name: 'Test Person' },
      { Name: 'Another Test', Age: '26', Position: 'Developer' },
      { Name: 'Third Test', Age: '19', Position: 'Salesperson' },
      { Age: '23', Name: 'End of this Page', Position: 'CEO', selected: true }
    ];

    const props = {
      data,
      renderSelector: selected => <div selected={selected} />
    };

    const wrapper = renderer.create(<Table {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render only children and ignore data and headers if childrens are present', () => {
    const data = [
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '23', Name: 'Lee Salminen' },
      { Age: '28', Position: 'Developer' },
      { Name: 'Griffin Smith', Age: '18' },
      { Age: '30', Name: 'Test Person' },
      { Name: 'Another Test', Age: '26', Position: 'Developer' },
      { Name: 'Third Test', Age: '19', Position: 'Salesperson' },
      { Age: '23', Name: 'End of this Page', Position: 'CEO', selected: true }
    ];

    const props = {
      data,
      renderSelector: selected => <div selected={selected} />
    };

    const wrapper = renderer
      .create(
        <Table {...props}>
          <div>
            WE ARE THE <span>CHILDREN!</span>
          </div>
        </Table>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  describe('interaction', () => {
    let sut;
    let data;
    let props;
    let onHeaderClick;
    let onDataClick;
    let onRowSelect;

    beforeEach(() => {
      data = [
        { Name: 'Griffin Smith', Age: '18' },
        { Age: '23', Name: 'Lee Salminen' },
        { Age: '28', Position: 'Developer' },
        { Name: 'Griffin Smith', Age: '18' },
        { Age: '30', Name: 'Test Person' },
        { Name: 'Another Test', Age: '26', Position: 'Developer' },
        { Name: 'Third Test', Age: '19', Position: 'Salesperson' },
        { Age: '23', Name: 'End of this Page', Position: 'CEO' }
      ];

      onHeaderClick = jest.fn();
      onDataClick = jest.fn();
      onRowSelect = jest.fn();

      props = { data, onHeaderClick, onDataClick, onRowSelect };
      sut = shallow(<Table {...props} />);
    });

    it('should allow to handle click on headers', () => {
      sut
        .find('[tid="header"]')
        .first()
        .simulate('click');
      expect(onHeaderClick).toHaveBeenCalledWith('Name', 0);
    });

    it('should allow to handle click on data', () => {
      sut
        .find('[tid="data"]')
        .last()
        .simulate('click');
      expect(onDataClick).toHaveBeenCalledWith(7, 2, 'Position', 'CEO');
    });

    it('should allow to select row', () => {
      sut
        .find('[tid="rowSelector"]')
        .last()
        .simulate('click');
      expect(onRowSelect).toHaveBeenCalledWith(data[7], 7);
    });
  });
});
