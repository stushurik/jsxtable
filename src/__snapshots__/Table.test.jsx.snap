// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table should allow to render row selector 1`] = `
<table>
  <thead>
    <tr
      className="headerRow"
    >
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Name
      </th>
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Age
      </th>
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Position
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        onClick={[Function]}
        tid="rowSelector"
      >
        <div
          selected={false}
        />
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        onClick={[Function]}
        tid="rowSelector"
      >
        <div
          selected={false}
        />
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Lee Salminen
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        onClick={[Function]}
        tid="rowSelector"
      >
        <div
          selected={false}
        />
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        28
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        onClick={[Function]}
        tid="rowSelector"
      >
        <div
          selected={false}
        />
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        onClick={[Function]}
        tid="rowSelector"
      >
        <div
          selected={false}
        />
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Test Person
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        30
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        onClick={[Function]}
        tid="rowSelector"
      >
        <div
          selected={false}
        />
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Another Test
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        26
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        onClick={[Function]}
        tid="rowSelector"
      >
        <div
          selected={false}
        />
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Third Test
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        19
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Salesperson
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        onClick={[Function]}
        tid="rowSelector"
      >
        <div
          selected={true}
        />
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        End of this Page
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        CEO
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`table should alow to render header with customization 1`] = `
<table>
  <thead>
    <tr
      className="headerRow"
    >
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        <span>
          Position
        </span>
      </th>
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Name
      </th>
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Age
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Lee Salminen
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        28
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Test Person
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        30
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Another Test
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        26
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Salesperson
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Third Test
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        19
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        CEO
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        End of this Page
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`table should alow to render row with customization 1`] = `
<table>
  <thead>
    <tr
      className="headerRow"
    >
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Name
      </th>
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Age
      </th>
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Position
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Lee Salminen
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        28
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Test Person
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        30
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Another Test
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        26
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Third Test
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        19
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Salesperson
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        <span>
          End of this Page
        </span>
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        <span>
          23
        </span>
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        <span>
          CEO
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`table should render only children and ignore data and headers if childrens are present 1`] = `
<table>
  <div>
    WE ARE THE 
    <span>
      CHILDREN!
    </span>
  </div>
</table>
`;

exports[`table should render table using custom tags 1`] = `
<myAwesomeTable>
  <myAwesomeThead>
    <myAwesomeTr
      className="headerRow"
    >
      <myAwesomeTh
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Name
      </myAwesomeTh>
      <myAwesomeTh
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Age
      </myAwesomeTh>
      <myAwesomeTh
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Position
      </myAwesomeTh>
    </myAwesomeTr>
  </myAwesomeThead>
  <myAwesomeTbody>
    <myAwesomeTr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </myAwesomeTr>
    <myAwesomeTr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Lee Salminen
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </myAwesomeTr>
    <myAwesomeTr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        28
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </myAwesomeTd>
    </myAwesomeTr>
    <myAwesomeTr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </myAwesomeTr>
    <myAwesomeTr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Test Person
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        30
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </myAwesomeTr>
    <myAwesomeTr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Another Test
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        26
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </myAwesomeTd>
    </myAwesomeTr>
    <myAwesomeTr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Third Test
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        19
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Salesperson
      </myAwesomeTd>
    </myAwesomeTr>
    <myAwesomeTr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        End of this Page
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </myAwesomeTd>
      <myAwesomeTd
        className="data"
        onClick={[Function]}
        tid="data"
      >
        CEO
      </myAwesomeTd>
    </myAwesomeTr>
  </myAwesomeTbody>
</myAwesomeTable>
`;

exports[`table should render table using default tags 1`] = `
<table>
  <thead>
    <tr
      className="headerRow"
    >
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Name
      </th>
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Age
      </th>
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Position
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Lee Salminen
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        28
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Test Person
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        30
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Another Test
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        26
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Third Test
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        19
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Salesperson
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        End of this Page
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        CEO
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`table should to set headers order 1`] = `
<table>
  <thead>
    <tr
      className="headerRow"
    >
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Position
      </th>
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Name
      </th>
      <th
        className="header"
        onClick={[Function]}
        tid="header"
      >
        Age
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Lee Salminen
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        28
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Griffin Smith
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        18
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      />
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Test Person
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        30
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Developer
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Another Test
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        26
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Salesperson
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        Third Test
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        19
      </td>
    </tr>
    <tr
      className="dataRow"
      onClick={[Function]}
      tid="rowSelector"
    >
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        CEO
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        End of this Page
      </td>
      <td
        className="data"
        onClick={[Function]}
        tid="data"
      >
        23
      </td>
    </tr>
  </tbody>
</table>
`;
